script/testing/util/db_server.py

```diff
@@ -227,21 +228,220 @@ def get_build_path(build_type):
+
+    Returns
+    -------
+    The preprocessed argument in the format expected by the DBMS.
+    """
+    # Currently do not require any attribute preprocessing
+    ATTR_PREPROCESSORS = []
+
+    # The preprocessing steps required for individual argument values
+    # NOTE(Kyle): The order of preprocessors is important here because,
+    # as one example, resolve_relative_paths looks for a relative path 
+    # designation (i.e. './') at the front of the value, while handle_flags
+    # will append the necessary '=' for non-flag arguments, which would 
+    # obviously confound relative path expansion if applied first.
+    VALUE_PREPROCESSORS = [
+        resolve_relative_paths,
+        lower_booleans,
+        handle_flags,
+    ]
+
+    preprocessed_attr  = apply_all(ATTR_PREPROCESSORS, attr, meta)
+    preprocessed_value = apply_all(VALUE_PREPROCESSORS, value, meta)
+    return f"-{preprocessed_attr}{preprocessed_value}"
+
+def applies_to(*target_types):
+    """
+    A decorator that produces a no-op function in the event that the 'target'
+    (i.e. first) argument provided to a preprocessing function is not an 
+    instance of one of the types specified in the decorator arguments.
 
-def generate_server_args_str(server_args):
-    """ Create a server args string to pass to the DBMS """
-    server_args_arr = []
-    for attribute, value in server_args.items():
-        value = str(value).lower() if isinstance(value, bool) else value
-        value = f'={value}' if value != None else ''
-        arg = f'-{attribute}{value}'
-        server_args_arr.append(arg)
+    This function should not be invoked directly; it is intended to be used
+    as a decorator for preprocessor functions to deal with the fact that 
+    certain preprocessing operations are only applicable to certain types.
+    """
+    def wrap_outer(f):
+        def wrap_inner(target, meta):
+            return f(target, meta) if any(isinstance(target, ty) for ty in target_types) else target
+        return wrap_inner
+    return wrap_outer
+
+# -----------------------------------------------------------------------------
+# Attribute Preprocessors
+#
+# The signature for attribute preprocessors must be:
+#   def preprocessor(attr: str, meta: Dict) -> str:
+#       ...
+
+
+# -----------------------------------------------------------------------------
+# Value Preprocessors
+#
+# The signature for value preprocessors must be:
+#   def preprocessor(value: str, meta: Dict) -> str:
+#       ...
+
+@applies_to(bool)
+def lower_booleans(value, meta):
+    """
+    Lower boolean string values to the format expected by the DBMS server.
+    
+    e.g. 
+        `True` -> `true`
+        `False` -> `false`
+    
+    Arguments
+    ---------
+    value : str 
+        The DBMS server argument value
+    meta : Dict
+        Dictionary of meta-information available to all preprocessors
+
+    Returns
+    -------
+    The preprocessed server argument value
+    """
```

Comment:
Want to add `assert value == True or value == False`?

Reply:
Assertion added.